src/App.tsx

```diff
@@ -1,14 +1,44 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [filter, setFilter] = useState('all');
+  const [query, setQuery] = useState('');
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+
+  useEffect(() => {
+    setLoading(true);
+    getTodos().then(setTodos).finally(() => setLoading(false));
+  }, []);
+
+  function getFilterTodos() {
+    return todos
+      .filter((todo) => todo.title.toLowerCase().includes(query.toLowerCase()))
+      .filter((todo) => {
```

Comment:
it is better to move filter inside switch case not vice verse to avoid an extra iteration when filter is `all`